biz.aQute.bndlib/src/aQute/bnd/build/model/BndEditModel.java

```diff
@@ -1111,22 +1196,73 @@ public void addBundleBlueprint(String location) {
+		List<String> includeResourceEntries1 = getEntries(Constants.INCLUDERESOURCE, listConverter);
+		List<String> includeResourceEntries2 = getEntries(Constants.INCLUDE_RESOURCE, listConverter);
+
+		return Stream.concat(includeResourceEntries1.stream(), includeResourceEntries2.stream())
+			.collect(toList());
 	}
 
-	public void setIncludeResource(List<String> includeResource) {
-		List<String> old = getIncludeResource();
-		doSetObject(aQute.bnd.osgi.Constants.INCLUDE_RESOURCE, old, includeResource, stringListFormatter);
+	public void setIncludeResource(List<String> newEntries) {
+		List<String> resourceEntries1 = getEntries(Constants.INCLUDERESOURCE, listConverter);
+		List<String> resourceEntries2 = getEntries(Constants.INCLUDE_RESOURCE, listConverter);
+
+		Set<String> resourceEntries = Stream.concat(resourceEntries1.stream(), resourceEntries2.stream())
+			.collect(toSet());
+
+		List<String> addedEntries = disjunction(newEntries, resourceEntries);
+		List<String> removedEntries = disjunction(resourceEntries, newEntries);
+
+		resourceEntries1.removeAll(removedEntries);
+		if (resourceEntries1.isEmpty()) {
+			removeEntries(Constants.INCLUDERESOURCE);
+		} else {
+			setEntries(resourceEntries1, Constants.INCLUDERESOURCE);
+		}
+
+		resourceEntries2.removeAll(removedEntries);
+		if (resourceEntries2.isEmpty()) {
+			removeEntries(Constants.INCLUDE_RESOURCE);
+		} else {
+			setEntries(resourceEntries2, Constants.INCLUDE_RESOURCE);
+		}
+
+		if (hasIncludeResourceInstruction()) {
+			resourceEntries1.addAll(addedEntries);
+			setEntries(resourceEntries1, Constants.INCLUDERESOURCE);
+		} else {
+			resourceEntries2.addAll(addedEntries);
+			setEntries(resourceEntries2, Constants.INCLUDE_RESOURCE);
+		}
 	}
 
 	public void addIncludeResource(String resource) {
-		List<String> includeResource = getIncludeResource();
-		if (includeResource == null)
-			includeResource = new ArrayList<>();
-		else
-			includeResource = new ArrayList<>(includeResource);
-		includeResource.add(resource);
-		setIncludeResource(includeResource);
+		String key = hasIncludeResourceHeader() ? Constants.INCLUDE_RESOURCE : Constants.INCLUDERESOURCE;
+		List<String> entries = getEntries(key, listConverter);
+		entries.add(resource);
+		setEntries(entries, key);
+	}
+
+	public void removeIncludeResource(String resource) {
+		List<String> resourceEntries1 = getEntries(Constants.INCLUDERESOURCE, listConverter);
+		List<String> resourceEntries2 = getEntries(Constants.INCLUDE_RESOURCE, listConverter);
+
+		if (resourceEntries1.contains(resource)) {
+			resourceEntries1.remove(resource);
+			setEntries(resourceEntries1, Constants.INCLUDERESOURCE);
+		}
+		if (resourceEntries2.contains(resource)) {
+			resourceEntries2.remove(resource);
+			setEntries(resourceEntries2, Constants.INCLUDE_RESOURCE);
+		}
+	}
+
+	private boolean hasIncludeResourceHeader() {
+		return objectProperties.containsKey(Constants.INCLUDE_RESOURCE);
```

Comment:
Shouldn't this be `properties` like hasIncludeResourceInstruction?

Reply:
No need for this method anymore. Removed.